mono/metadata/exception-internals.h

```diff
@@ -71,4 +71,49 @@ mono_exception_new_thread_abort (MonoError *error);
 MonoExceptionHandle
 mono_exception_new_thread_state (const char *msg, MonoError *error);
 
+void
+mono_set_pending_exception_class_failure (MonoClass *klass);
+
+void
+mono_set_pending_exception_remoting (const char *message);
+
+void
+mono_set_pending_exception_appdomain_unloaded (void);
+
+void
+mono_set_pending_exception_argument (const char *name, const char *message);
+
+void
+mono_set_pending_exception_argument_null (const char *name);
+
+void
+mono_set_pending_exception_argument_out_of_range (const char *name);
+
+void
+mono_set_pending_exception_divide_by_zero (void);
+
+void
+mono_set_pending_exception_execution_engine (const char *message);
+
+void
+mono_set_pending_exception_index_out_of_range (void);
+
+void
+mono_set_pending_exception_invalid_operation (const char *message);
+
+void
+mono_set_pending_exception_not_supported (const char *message);
+
+void
+mono_set_pending_exception_overflow (void);
+
+void
+mono_set_pending_exception_synchronization_lock (const char *message);
+
+void
+mono_set_pending_exception_thread_interrupted (void);
+
+void
+mono_set_pending_exception_not_implemented (const char *message);
+
```

Comment:
We are trying to discourage the implicit use of the pending exception in favor of explicitly threading `MonoError*` through the code - rather than adding a dozen new pending exception functions, I think we should be trying to replace uses of `mono_set_pending_exception (mono_exception_get_FOO(...))` by `mono_error_set_FOO (error, ...)` followed by a `mono_error_set_pending_exception(error)` or by routing the mono-error out of a `HANDLES()` icall.

Reply:
I think that can come later but it is also optional I believe. This is an easy effective mechanical change. For example if the functions don't have pointers, this suffices. And if they gain handles() and MonoError, this still works. Other than removing a small amount of code, what is the advantage of the other approach?